### DISCUSSION TITLE
Possible Jak II/3 Enhancement Ideas (For When They're Stable Enough)

### DISCUSSION THREAD

**Discussion by BakerIsHungry** [open-goal/jak-project#2501] [Ideas]
### Is your feature request related to a problem?

1. When you get to the title screen, only the beginning of the Jak II theme plays, then the music just cuts to silence. The full song by Josh Mancell is two minutes and 14 seconds long.

2. Haven City's layout is considerably different in Jak 3 than in Jak II.

3. The gunplay is whack.

### Describe the solution you'd like.

1. If possible, if the entire track is still located in the files, maybe consider including the full theme in the title screen? That'd be a pretty awesome surprise for returning players.

2. Not sure if it's possible without having to change too much of the game to make it all function properly, but maybe redesign the layout of Jak 3 to more closely resemble Jak II? It'd be pretty cool to see Jak II and 3 share one Haven City layout, but that might be too much work in the long run because then people would want the entire game redesigned to match.

3. Jak II and 3 have cool weapons, but the gunplay feels like it would've been so much better with a lock strafe control scheme for when your weapons are out. If possible, maybe make the firing controls function more like Ratchet: Deadlocked? 

### Additional context

1. https://soundcloud.com/joshmancell/jak2-renegade-main-theme

2. 
![Haven_City_map_from_Jak_II](https://user-images.githubusercontent.com/130881357/232348602-82aba353-eb7f-4ec2-bb08-cf01f38b1264.png) 
![Haven_City_map_from_Jak_3](https://user-images.githubusercontent.com/130881357/232348611-7a4272b5-e795-4dc7-a2d4-bc180951e4d0.png)

3. 
![6253_2004-03-22_12-723898-705081860](https://user-images.githubusercontent.com/130881357/232349015-3d18e0d5-0a21-4394-9048-6ef27de4f43e.jpg) 
![f39d1d65-775d-4b0b-8bd9-db9440b06e2d-2203922297](https://user-images.githubusercontent.com/130881357/232349024-e01729e6-83fd-4df2-9018-3cafd3dd12b7.jpg)

**Comment by BakerIsHungry** [comment]
Sorry if I put in the incorrect label btw. I'm new to Github, so I wasn't sure how to change it to enhancement.

**Reply by xTVaser** [reply]
It's fine, I just enabled the discussions feature for the first time (this is our first discussion).  Ideally issues are reserved for tasks we are definitely going to do / bugs we have to fix.  But this is a perfect example of an idea discussion -- that may have issues generated from it.

During jak 1 we had a huge issue for similar ideas that got way out of hand

**Reply by BakerIsHungry** [reply]
Understood. I wasn't sure whether this or the Discord was a better place for ideas such as these, but I chose this because I like to keep my server list small and full of ones I usually interact with. Plus knowing how much of a pain it must've been to translate GOAL in the first place, I could only imagine ideas like this were kinda whacky to begin with. Still worth mentioning though, I think.

**Comment by trippjoe** [comment]
I'd love to hear more about the second bullet point. Could you possibly draw up a sketch of your ideas for rearranging Haven City?

**Comment by Gorilli09** [comment]
L2 can be used to Strafe and pressing L2 & R2 simultaneously engages Dark Jak. Also, creating strafe animations would be very hard to do, which isn't likely going to happen.

**Reply by BakerIsHungry** [reply]
Coulda sworn I've seen Jak do a kind of strafe-ish animation with the minigun. Like his torso turns to whatever enemy he's locked on? Maybe that could be used for said strafing animation, but idk.

**Comment by doctashay** [comment]
Haven City changes between 2 and 3 are tricky. The Jak 3 July internal build is actually compatible with Jak 2 levels, so even while OpenGOAL doesn't support Jak 3 yet, I've played around with bringing the full industrial (`ctyinda`/`ctyindb`) and "gen" areas (by the stadium, `ctygena`/`ctygenb`) back to the Jak 3 city. 

Here are some screenshots from that modded internal build I took about two years ago:

![image](https://github.com/open-goal/jak-project/assets/70249609/b320a141-b5de-453d-b645-267e2a52be44)

![image](https://github.com/open-goal/jak-project/assets/70249609/972c0cef-5e28-45fb-a0d4-f961dbfa04fa)

![image](https://github.com/open-goal/jak-project/assets/70249609/c69df4d7-a88a-483b-b640-2fb5667a1748)

![image](https://github.com/open-goal/jak-project/assets/70249609/80d1c5b1-f196-4445-9c98-3cb8bfecb89b)

Eventually, I'd love to see a mod where the city gets progressively rebuilt once the game is completed. I always thought it's weird how you defeat Cyber-Errol, effectively ending both the Metal Head and KG threats, yet the city stays deadlocked in this perpetual war that never ends.
